### PR TITLE
feat(3416039642): Add "Connect Wallet" functionality

### PR DESCRIPTION
**Pull request by galuzan-starkware**
### Description of the Changes

- Adding WalletButton to the header
- Adding Styling to Align WalletButton to be with the same font and margin
- Adding autoConnect env var
- Changing build node version to 16
- Adding `envs.js` file, export REACT_APP envs

Solves #[3416039642](https://starkware.monday.com/boards/2847513897/pulses/3416039642)

---

### Checklist

- [ ] Manually tests of the main Application flows are done and passed.
- [ ] New unit / functional tests have been added (whenever applicable).
- [ ] Docs have been updated (whenever relevant).
- [ ] PR title is follow the [Conventional Commits](https://www.conventionalcommits.org/) convention: `<type>[optional scope]: <description>`, e.g: `fix: prevent racing of requests`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,15 +42,23 @@
...
     "react-dom": "^17.0.2",
     "react-scripts": "4.0.3",
-    "react-split-pane": "^2.0.3"
+    "react-split-pane": "^2.0.3",
+    "starknet": "^4.12.0"
   },
   "devDependencies": {
+    "@emotion/react": "^11.10.5",
+    "@emotion/styled": "^11.10.5",
+    "@mui/material": "^5.10.12",
```
Comment (dan-ziv) [inline package.json]: Why this is in the `devDependencies`?

File: src/constants/links.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const Links = {
```
Comment (dan-ziv) [inline src/constants/links.js]: Why wrap this with `Links` object?
We can export directly the link var:
`export const CONTRACT_URL = ....`

File: src/constants/links.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+  CONTRACT_URL: 'https://goerli.voyager.online/contract/',
```
Comment (dan-ziv) [inline src/constants/links.js]: Why voyager and not starkscan?

**Review by dan-ziv — CHANGES_REQUESTED** [review]

File: src/constants/links.js (hunk 2 above)
Comment (galuzan-starkware) [reply src/constants/links.js]: I align it with the current const design. Check https://github.com/starkware-libs/cairo-playground/blob/2127c9a61bb506bef188077c07429c114b2db0e0/src/constants/action-types.js

File: src/constants/links.js (hunk 3 above)
Comment (galuzan-starkware) [reply src/constants/links.js]: I asked you yesterday, and you told me to leave it with voyager

File: package.json (hunk 1 above)
Comment (galuzan-starkware) [reply package.json]: moved it to dependencies

File: src/constants/links.js (hunk 2 above)
Comment (dan-ziv) [reply src/constants/links.js]: This is not the same, those are enums-like

File: src/constants/links.js (hunk 3 above)
Comment (dan-ziv) [reply src/constants/links.js]: To leave what? you added it as new URLs

Comment (galuzan-starkware) [reply src/constants/links.js]: https://github.com/starkware-libs/cairo-playground/blob/6ea320a859c0769c5035771c410896d4dbe6c11c/src/components/Features/Output/DeployOutput.js#L4
they were here. I just changed the location cause I needed `CONTRACT_URL` in `StarknetWalletButton`

File: src/constants/links.js (hunk 2 above)
Comment (galuzan-starkware) [reply src/constants/links.js]: done

**Review by dan-ziv — COMMENTED** [review]
@galuzan-starkware please fix build

File: src/constants/links.js (hunk 3 above)
Comment (dan-ziv) [reply src/constants/links.js]: OK, if this is just changing this so lets change it to starkscan please

Comment (galuzan-starkware) [reply src/constants/links.js]: done

**Review by dan-ziv — APPROVED** [review]

**Pull request merged**